### ISSUE TITLE
TypeScript : TWEEN.Tween could not be used as a type of variables

### ISSUE THREAD

**Issue by MasatoMakino** [tweenjs/tween.js#559]
## Expected Behavior

`TWEEN.Tween` can be used as a type of variables.

## Current Behavior

`TWEEN.Tween` could not be used as a type of variables.

## Possible Solution

It is not happening in tween.js 18.5.0 + @types/tween.js 18.5.1

## Steps to Reproduce

Reproduction Repository
https://github.com/MasatoMakino/behavior-type-Tween.tween.git

1. clone repository
2. npm run tsc

error log

```
$ tsc src/TestTypeOfTween.ts --esModuleInterop
src/TestTypeOfTween.ts:4:22 - error TS2503: Cannot find namespace 'TWEEN'.

4 const tweenArray01 : TWEEN.Tween[] = []; //NG
                       ~~~~~

src/TestTypeOfTween.ts:5:22 - error TS2749: 'Tween' refers to a value, but is being used as a type here. Did you mean 'typeof Tween'?

5 const tweenArray02 : Tween[] = [];  //NG
                       ~~~~~

src/TestTypeOfTween.ts:8:15 - error TS2503: Cannot find namespace 'TWEEN'.

8 const tween02:TWEEN.Tween = new TWEEN.Tween({x:1.0}); //NG
                ~~~~~

src/TestTypeOfTween.ts:9:15 - error TS2749: 'Tween' refers to a value, but is being used as a type here. Did you mean 'typeof Tween'?

9 const tween03:Tween = new TWEEN.Tween({x:1.0}); //NG
                ~~~~~


Found 4 errors.

error Command failed with exit code 2.
info Visit https://yarnpkg.com/en/docs/cli/run for documentation about this command.

Process finished with exit code 2
```

## Environment

    "@tweenjs/tween.js": "18.6.0",
    "typescript": "^3.9.7"

Thanks.

**Comment by andrewkolos** [comment]
I believe this is because `TWEEN.Tween` isn't actually the Tween class, it is only the constructor (thus we lose the typing). The same applies to the rest of the types.
If we look at `index.ts`, `Tween` is never exported:
```TypeScript
import NOW from './Now'
// ...
import Tween from './Tween'
// ...

class Main extends Group {
	public version = VERSION
	public now = NOW
	public Group = Group
	public Easing = Easing
	public Interpolation = Interpolation
	public nextId = Sequence.nextId
	public Tween = Tween
}

const TWEEN = new Main()
export default TWEEN
```
The way I thought to fix this without breaking the existing API is by exporting all the types after the default export. For that I just add the following to `index.ts`
```TypeScript
export { Group, Easing, Interpolation, Sequence, Tween }
```
However, looking at the build scripts, I believe doing this wouldn't change anything as the build just emits a namespace declaration for the default export `TWEEN` as a single `outFile` taken and doesn't emit typedef files for each of the types. I made [a fork ](https://github.com/andrewkolos/tween.js) ([diff](https://github.com/andrewkolos/tween.js/commit/2c99e17f152dd1d050d4679398bc4343b4668440)) that attempts to address this, but I get a warning from rollup: 
```
(!) Mixing named and default exports
Consumers of your bundle will have to use bundle['default'] to access the default export, which may not be what you want. Use `exports: 'named'` to disable this warning
```
I'm not very familiar with all the non ES-standard JS module systems, so I'm not sure what this means exactly. I'm assuming mixing default exports with named ones isn't good. I am not sure how to address this while still solving the problem. In addition, something I did broke a lot of the unit tests. I am not sure what went wrong and am out of time to investigate--sorry.

**Comment by trusktr** [comment]
PR welcome!

Also, an easy fix is instead of writing

```js
import * as TWEEN from '@tweenjs/tween.js'

new TWEEN.Tween
// ...
TWEEN.Easing
```

just write

```js
import {Tween, Easing} from '@tweenjs/tween.js'

new Tween
// ...
Easing
```

In that case the types work fine.

**Comment by trusktr** [comment]
I think we should also move away from `namespace`.

**Comment by chriso86** [comment]
How are you not getting the errors below? Does this library work with TypeScript? I've only had frustration so far.

![image](https://user-images.githubusercontent.com/15611519/95657260-cdcd4c80-0b13-11eb-9703-d339b00de623.png)

**Comment by trusktr** [comment]
@chriso86 I see some errors in your screenshot say `imported as 'TWEEN'` which I think may indicate you are importing like `import * as TWEEN from '@tweenjs/tween.js'` which noted above doesn't currently work in TypeScript.

I haven't had that issue when importing it like `import {Tween, Easing} from '@tweenjs/tween.js'`.

Did you try making sure you import directly the things you need, instead of `* as TWEEN` for now?

**Comment by trusktr** [comment]
Another workaround, for now, is to pin it to 18.5.0 (without types). I'll aim to fix this ASAP unless someone else can get to it first.

**Comment by chriso86** [comment]
Hi Joe,

I've tried that, alas to no avail.

Thanks for your response.

I'll wait for the next version release, update and try again.

Thanks again.

**Comment by trusktr** [comment]
Related to https://github.com/tweenjs/tween.js/issues/566. We will need to fix the output so that the typedef matches with the ESM output file, and update package.json accordingly. This means it may start working for most of us, and then break for some people still using older versions of tools that expect CommonJS, but this is the way forward.

**Comment by trusktr** [comment]
Got this fixed, but no more time to polish up the changes tonight. Will finish and release it soon, hopefully tomorrow (Monday).

**Comment by trusktr** [comment]
Got the PR up at https://github.com/tweenjs/tween.js/pull/568

Basically it works the same in old CommonJS/UMD/AMD setups, but now for ES Modules (and TypeScript) it works like this:

```js
import {Twean, Easing, ...} from '@tweenjs/tween.js'
```

There is no more `default` export, so if you want the `TWEEN` object as before, then import like this:

```js
import * as TWEEN from '@tweenjs/tween.js'
```

The new implementation is the idiomatic way to write ES Modules that are compatible with CommonJS modules without hacks (f.e. without merging `default` and named exports into a single CommonJS `module.exports` object).

**Comment by trusktr** [comment]
Done and released in `18.6.1`. Please give it a try and let me know if any issues. It should work in any format now:

```js
import TWEEN from '@tweenjs/tween.js'
```
```js
import * as TWEEN from '@tweenjs/tween.js'
```
```js
import {Tween, Easing, ...} from '@tweenjs/tween.js'
```

plus the same as before in CommonJS, AMD, or UMD.
